server/reports/attendance/attendance-endpoint.js

```diff
@@ -73,64 +67,62 @@ export function attendanceRequestHandler(request, response) {
   MindbodyAccess.getAuth()
     .then((value) => {
       MindbodyAccess.authToken = value.AccessToken;
-      // gets all classes between StartDateTime - EndDateTime
-      return MindbodyAccess.getClasses({
-        StartDateTime: startdate,
-        EndDateTime: enddate,
-      });
+      return MindbodyAccess.getClasses();
     })
     // gets all classes from "Classes" endpoint
     .then((classes) => {
-      console.log("Classes Pagination Data: ", classes.PaginationResponse); // shows pagination data in console
-      let attendanceReport = [];
+      let attendanceReport = {
+        data: [],
+        headers: ["class", "capacity", "registered", "attended"],
+      };
       let allNumberAttendedPromises = [];
       let numberOfClasses = Object.keys(classes.Classes).length;
 
       // iterates through every class in classes
       for (let i = 0; i < numberOfClasses; ++i) {
-        let classTitle = classes.Classes[i].ClassDescription.Name;
         let classID = classes.Classes[i].Id;
         let maxCapacity = classes.Classes[i].MaxCapacity;
         let numberRegistered = classes.Classes[i].TotalBooked;
-        //let classDate = fixDateFormat(classes.Classes[i].StartDateTime); // Mindbody StartDateTime format is yyyy/mm/dd by default
+        let classDate = fixDateFormat(classes.Classes[i].StartDateTime); // Mindbody StartDateTime format is yyyy/mm/dd by default
+
+        // adds class data to attendance report if startdate <= classdate <= enddate
+        if (isValidDate(startdate, classDate, enddate)) {
+          let classData = {
+            class: classID,
+            capacity: maxCapacity,
+            registered: numberRegistered,
+            attended: 0
+          };
```

Comment:
I'm so sorry I understand now. This is a huge mistake on my part. I wrote the fixDateFormat and isValidDate functions a while back when I misunderstood how to get classes between the startdate and enddate. I don't use them intentionally in the current state of dev and honestly thought I removed those functions from my code all together. To get all classes between the startdate and enddate, you need to revert the code I commented on above between lines -77 to +70. So the getClasses function call should look like:

      return MindbodyAccess.getClasses({
        StartDateTime: startdate,
        EndDateTime: enddate,
      });